modules/images/webp-uploads/load.php

```diff
@@ -154,6 +182,26 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 				continue;
 			}
 
+			$original_thumbnail_image_filesize = isset( $properties['filesize'] ) ? (int) $properties['filesize'] : 0;
+			$webp_thumbnail_image_filesize     = isset( $source['filesize'] ) ? (int) $source['filesize'] : 0;
+			if ( $original_thumbnail_image_filesize <= 0 && $webp_thumbnail_image_filesize <= 0 ) {
+				continue;
+			}
+
+			/** This filter is documented in modules/images/webp-uploads/load.php */
+			if (
+				apply_filters( 'webp_prioritise_filesize_over_mime_type', true )
+				&& $webp_thumbnail_image_filesize >= $original_thumbnail_image_filesize
+			) {
+				$destination = trailingslashit( $original_directory ) . "{$source['file']}";
```

Comment:
Use `path_join` instead of `trailingslashit`:
```suggestion
				$destination = path_join( $original_directory, $source['file'] );
```

Reply:
Nice catch. I followed a similar format that is used here: https://github.com/WordPress/performance/blob/trunk/modules/images/webp-uploads/load.php#L92 But this review makes more sense than the current approach.